### ISSUE TITLE
URL slug not staying as expected after scroll of @scalar/api-reference documention

### ISSUE THREAD

**Issue by Rahul-kalsha** [scalar/scalar#1339]
### What happens?

I integrated scalar API documentation via [CDN link](https://github.com/scalar/scalar?tab=readme-ov-file#from-a-cdn) in angular SPA project. but there is a issue coming on url slug. my existing url slug not staying when scroll down the API documentation. Here i am adding the code. Help me to fix this bug. 

Here is the component file:

```typescript
import { AfterViewInit, Component, OnDestroy, OnInit, Renderer2 } from '@angular/core';

@Component({
  selector: 'app-api-list',
  standalone: true,
  imports: [],
  templateUrl: './api-list.component.html',
  styleUrls: ['./api-list.component.scss'],
})
export class ApiListComponent implements OnInit, AfterViewInit, OnDestroy {
  constructor() { }

  ngOnInit(): void {
    this.loadScript('https://cdn.jsdelivr.net/npm/@scalar/api-reference').then((res: any) => {
      console.log('res:', res);
    }).catch(error => {
      console.error('Error loading script:', error);
    });
  }

  ngOnDestroy(): void {
  }

  ngAfterViewInit(): void {
  }

  private loadScript(url: string): Promise<void> {
    return new Promise<void>((resolve, reject) => {
      const scriptElement = document.createElement('script');
      scriptElement.src = url;
      scriptElement.onload = () => resolve();
      scriptElement.onerror = (error) => reject(error);
      document.body.appendChild(scriptElement);
    });
  }
}
``` 

Here is the html file:
```html
<div
id="api-reference"
data-url="https://petstore3.swagger.io/api/v3/openapi.json"
data-proxy-url="https://api.scalar.com/request-proxy"></div>
``` 

You can check visually here and suggest me proper solution if i am miss any configuration then also please help me.

https://github.com/scalar/scalar/assets/56481473/de108e22-09c8-4322-8d73-19c3837b78b0

### What did you expect to happen?

When we scroll down API documentation Url 

Should be like:

```
http://localhost:4200/public-api-list#tag/pet
http://localhost:4200/public-api-list#tag/store
http://localhost:4200/public-api-list#tag/user
http://localhost:4200/public-api-list#models
```

Not like that:

```
http://localhost:4200/#tag/pet
http://localhost:4200/#tag/store
http://localhost:4200/#tag/user
http://localhost:4200/#models
```

### How can we reproduce the issue?

1. Setup CDN in Angular project as i already mentioned code above
2. Setup API documentation page in particular url slug 
3. Scroll down to API documention page
4. See URL slug

**Comment by marclave** [comment]
@Rahul-kalsha thanks for reporting this bug! and all the added details!

@amritk for our router should we be ensuring the root path is kept?

on an aside, @amritk should we make an Angular integration :)

**Comment by amritk** [comment]
Hey yea this should be a pretty easy fix!

**Comment by amritk** [comment]
And also yes haha

**Comment by amritk** [comment]
Alright PR is up https://github.com/scalar/scalar/pull/1348

**Comment by Rahul-kalsha** [comment]
Hi @amritk  and @marclave, All anchor tags on the sidebar should also be the same

http://localhost:4200/public-api-list#tag/pet
http://localhost:4200/public-api-list#tag/store
http://localhost:4200/public-api-list#tag/user
http://localhost:4200/public-api-list#models 

![image](https://github.com/scalar/scalar/assets/56481473/ca4e2eaf-090c-445d-86db-c7e46b3b0d63)

**Comment by marclave** [comment]
fixed in `@scalar/api-reference@1.20.15` :)

**Comment by Rahul-kalsha** [comment]
Thank you @marclave and @amritk, for fixing this issue. Good work and it will be very helpful.
